default/libraries/ssh_crypto.rb

```diff
@@ -0,0 +1,148 @@
+# encoding: utf-8
+#
+# Copyright 2015, Dominik Richter
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# author: Christoph Hartmann
+# author: Dominik Richter
+# author: Patrick Muench
+
+class SshCrypto
+  attr_reader :os
+  def initialize(os)
+    @os = os
+  end
+
+  def valid_ciphers
+    # define a set of default ciphers
+    ciphers53 = 'aes256-ctr,aes192-ctr,aes128-ctr'
+    ciphers66 = 'chacha20-poly1305@openssh.com,aes256-gcm@openssh.com,aes128-gcm@openssh.com,aes256-ctr,aes192-ctr,aes128-ctr'
+    ciphers = ciphers53
+
+    # adjust ciphers based on OS + release
+    case os[:family]
+    when 'Ubuntu'
```

Comment:
this needs to be 'ubuntu', the same applies for 'debian' and 'redhat'. Once we have this, I am going to merge it